### PR TITLE
Add support for the Hackable Electronic Badge

### PR DESCRIPTION
**Pull request by DarkmatterVale**
Since I am not making any major changes to the backend of BlocklyProp, we can merge this branch back into master.

### CODE REVIEW

File: src/main/webapp/cdn/framec.html [hunk 1]
```diff
@@ -153,6 +154,7 @@
                 <block type="serial_tx_byte"></block>
                 <block type="serial_send_text"></block>
                 <block type="serial_rx_byte"></block>
+                <block type="string_type_block"></block>
```
Comment (michel-cf) [inline src/main/webapp/cdn/framec.html]: This block does not follow the naming scheme of the other blocks. Please change

**Comment by michel-cf** [comment]
src/main/webapp/cdn/framec.html is not in use anymore. As this page contains the texts for the categories in the toolbox they moved to src/main/webapp/frame/framec.jsp to support translations.

Many of the blocks have names that come close to blocks that might once also be available. please prepend all blocks with `heb_`.

**Comment by DarkmatterVale** [comment]
@michel-cf Is the PR good now?

**Comment by DarkmatterVale** [comment]
Can I merge this in?

**Comment by michel-cf** [comment]
While unused, the string_type_block is still present and the block categories are not translatable.

Looks good for the rest.

**Comment by DarkmatterVale** [comment]
I have reimplemented the string_type_block and made the categories translatable.

**Pull request merged**